web/src/lib/roadmap/roadmapBuilder.ts

```diff
@@ -128,37 +131,32 @@ const findTaskInRoadmapByFilename = (
   roadmapBuilder: RoadmapBuilder,
   taskFilename: string
 ): TaskBuilder | undefined => {
-  for (const step of roadmapBuilder.steps) {
-    const found = step.tasks.find((task) => task.filename === taskFilename);
-    if (found) return found;
-  }
+  const found = roadmapBuilder.tasks.find((task) => task.filename === taskFilename);
+  if (found) return found;
 };
 
 const convertToRoadmap = async (roadmapBuilder: RoadmapBuilder): Promise<Roadmap> => {
   const roadmap = {
-    steps: await Promise.all(
-      roadmapBuilder.steps.map(async (step) => ({
-        ...step,
-        tasks: await Promise.all(
-          step.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
-        ),
-      }))
+    steps: await Promise.all(roadmapBuilder.steps),
+    tasks: await Promise.all(
+      roadmapBuilder.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
     ),
   };
 
-  const allFilenames = roadmap.steps.reduce(
-    (acc: string[], currStep: Step) => [...acc, ...currStep.tasks.map((task) => task.filename)],
-    []
+  roadmap.tasks.map(
+    (task: Task) =>
+      (task.stepNumber = (
+        findTaskInRoadmapByFilename(roadmapBuilder, task.filename) as TaskBuilder
+      ).step_number)
```

Comment:
what is this line doing? we're not saving the return value of this map...

Reply:
yeah, I'm not sure. It's not doing anything, I don't where it even came from, I will get rid of it.

Reply:
so after some digging, this line is updating the step numbers within in roadmap but I'm not sure why it is because as you said we are not saving the values?

Reply:
I figured it out, I was using the map as for loop to traverse and assign step numbers so I just made it a for of loop so we don't waste memory.